sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -150,6 +150,57 @@ object IntervalUtils {
     }
   }
 
+  private val daySecondStringPattern = ("(?i)^(INTERVAL\\s+)([+|-])?(')" +
+    "([+|-])?(\\d+) (\\d{1,2}):(\\d{1,2}):(\\d{1,2})(\\.\\d{1,9})?(')(\\s+DAY\\s+TO\\s+SECOND)$").r
+  private val daySecondPattern = "^([+|-])?(\\d+) (\\d{1,2}):(\\d{1,2}):(\\d{1,2})(\\.\\d{1,9})?$".r
+
+  def castStringToDTInterval(input: UTF8String): Long = {
+    def secondAndMicro(second: String, micro: String): String = {
+      if (micro != null) {
+        s"$second$micro"
+      } else {
+        second
+      }
+    }
+    val intervalStr = input.trimAll().toString
+    intervalStr match {
+      case daySecondPattern("-", day, hour, minute, second, micro) =>
+        toDTInterval(day, hour, minute, secondAndMicro(second, micro), -1)
+      case daySecondPattern(_, day, hour, minute, second, micro) =>
+        toDTInterval(day, hour, minute, secondAndMicro(second, micro), 1)
+      case daySecondStringPattern(
+      _, firstSign, _, secondSign, day, hour, minute, second, micro, _, _) =>
```

Comment:
You don't need to put quotes to groups:
```suggestion
      case daySecondLiteralPattern(firstSign, _, secondSign, day, hour, minute, second, micro) =>
```
if you change the pattern to:
```
  private val daySecondLiteralPattern =
    (s"(?i)^(INTERVAL\\s+)([+|-])?\\'$unquotedDaySecondPattern\\'\\s+DAY\\s+TO\\s+SECOND$$").r
```